### PR TITLE
Create new React frontend for Nebari

### PR DESCRIPTION
**Pull request by jbouder**
## What does this implement/fix?

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds a feature)
- [x] Breaking change (fix or feature that would cause existing features not to work as expected)
- [ ] Documentation Update
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [ ] Other (please describe):

## Testing

- [x] Did you test the pull request locally?
- [x] Did you add new tests?

## Documentation

- Added README to `ui` directory with instructions on running locally, development, and testing

## Screenshots
Screenshot 2023-11-17 at 11 49 42 AM
Screenshot 2023-11-17 at 11 49 54 AM

## Related Issues:
https://github.com/nebari-dev/jhub-apps/issues/13
https://github.com/nebari-dev/jhub-apps/issues/14

## Any other comments?
- This update includes the base homepage updates created in React
- This update does not include the following React UI functionality: 
  - Create App, currently the existing panel app creation form is used
  - Edit, delete, start, stop functionality
  - Search functionality
- Future styling is to be expected

### CODE REVIEW

File: jupyterhub_config.py [hunk 1]
```diff
@@ -1,13 +1,110 @@
...
+            "name": "JuypterLab",
+            "url": hub_url,
+            "display": True,
+            "info": {
+                "name": "JupyterLab",
+                "url": "/user/[USER]/lab",
+                "external": True,
+            },
+            "oauth_no_confirm": True,
+        },
```
Comment (jbouder) [inline jupyterhub_config.py:34]: Currently the Service buttons are populated with default JupyterHub services. These are added to provide a template for adding more as needed

File: jupyterhub_config.py [hunk 2]
```diff
@@ -1,13 +1,110 @@
...
 c.JupyterHub.authenticator_class = DummyAuthenticator
 c.JupyterHub.log_level = 10
 
-c.JupyterHub.bind_url = "http://127.0.0.1:8000"
+hub_url = "http://127.0.0.1:8000"
+
+c.JupyterHub.bind_url = hub_url
 c = install_jhub_apps(c, spawner_to_subclass=SimpleLocalProcessSpawner)
+
+c.JupyterHub.default_url = "/hub/home"
```
Comment (jbouder) [inline jupyterhub_config.py:19]: This ensures that a user is redirected to the homepage when they click the top right logo, are are directed to the default_url by some JupyterHub mechanism

**Review by aktech — CHANGES_REQUESTED** [review]

File: .vscode/extensions.json [hunk 3]
```diff
@@ -0,0 +1,10 @@
+{
```
[the remark this answers is not in the document]
Comment (jbouder) [reply .vscode/extensions.json:1]: Removed

File: jhub_apps/templates/page.html [hunk 4]
```diff
@@ -0,0 +1,70 @@
+<!-- Copyright (c) Nebari Development Team.-->
+<!-- Distributed under the terms of the Modified BSD License. -->
+
+{% extends "templates/page.html" %} {% block nav_bar_left_items %}
+
+<!-- Default nav bar items -->
+<li>
+  <a href="{{base_url}}home {% if cdsdashboards_restricted %}-cds{% endif %}"
```
[the remark this answers is not in the document]
Comment (jbouder) [reply jhub_apps/templates/page.html]: This file was copied from the original nebari page template, so it was a carry over. Removed as it is no longer applicable.

File: ui/README.md [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+2. To start the app, run the following:
+
+```sh
+npm run dev
+```
+
+3. To load the UI, navigate to the following:
+
+```
+http://localhost:8080/hub/home
```
[the remark this answers is not in the document]
Comment (jbouder) [reply ui/README.md:29]: Its appears to be loading fine for me...will contact offline.

**Pull request merged**